File: nodejs/src/day0.js
const fs = require("fs");

module.exports = function () {
  for (let i = 0; i < 1000; i++) {
    p1();
    p2();
  }
};

function p1() {
  const data = fs.readFileSync("../data/day0.txt", "utf8");

  let prev = -1;
  let count = 0;
  data.split("\n").forEach((el) => {
    el = parseInt(el);
    if (prev >= 0 && prev < el) {
      count += 1;
    }
    prev = el;
  });

  // console.log("Part 1: " + count);
}

function p2() {
  const data = fs.readFileSync("../data/day0.txt", "utf8");

  const ee = [];
  data.split("\n").forEach((el, i) => {
    ee[i] = parseInt(el);
  });

  let count = 0;
  for (let i = 0; i < ee.length - 4; i++) {
    if (ee[i] + ee[i + 1] + ee[i + 2] < ee[i + 1] + ee[i + 2] + ee[i + 3]) {
      count += 1;
    }
  }

  // console.log("Part 2: " + count);
}

File: ruby/day0.rb
def day0
  1000.times do
    p1
    p2
  end
end

def p1
  data = File.read('../data/day0.txt')

  prev = -1
  count = 0
  data.split("\n") do |el|
    el = el.to_i
    if prev >= 0 && prev < el
      count += 1
    end
    prev = el
  end

  # puts "Part 1: #{count}"
end

def p2
  data = File.read('../data/day0.txt')

  ee = []
  data.split("\n").each.with_index do |el, i|
    ee[i] = el.to_i
  end

  count = 0
  for i in 0..(ee.length - 4)
    if (ee[i] + ee[i + 1] + ee[i + 2]) < (ee[i + 1] + ee[i + 2] + ee[i + 3])
      count += 1
    end
  end

  # puts "Part 2: #{count}"
end
